diff --git a/camlibs/ptp2/chdk.c b/camlibs/ptp2/chdk.c
index e1d898f37..90f884e42 100644
--- a/camlibs/ptp2/chdk.c
+++ b/camlibs/ptp2/chdk.c
@@ -1150,7 +1150,7 @@ static void yuv_live_to_jpeg(unsigned char *p_yuv,
 	struct		jpeg_error_mgr jerr;
 	JSAMPROW	row_ptr[1];
 	uint8_t		*outbuf = NULL, *tmprowbuf = NULL;
-	uint64_t	outlen = 0;
+	unsigned long	outlen = 0;
 	unsigned int	row_inc;
 	int		sshift, dshift, xshift, skip;
 
